CMakeLists.txt

```diff
@@ -16,24 +16,30 @@ set(GCF_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/git-cmake-format.py)
 
 find_package(ClangFormat ${GCF_CLANGFORMAT_MINIMAL_VERSION} REQUIRED)
 
-execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --absolute-git-dir
+execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --git-path hooks
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  OUTPUT_VARIABLE GCF_GIT_ROOT
+  OUTPUT_VARIABLE GCF_GIT_HOOKS_DIR
   OUTPUT_STRIP_TRAILING_WHITESPACE)
-if(NOT GCF_GIT_ROOT)
+if(NOT GCF_GIT_HOOKS_DIR)
   message(WARNING "Not in a git repository")
 else()
-  set(GCF_GIT_DIR_SCRIPT ${GCF_GIT_ROOT}/hooks/git-cmake-format.py)
-  configure_file(
-    ${GCF_SCRIPT} ${GCF_GIT_DIR_SCRIPT}
-    COPYONLY)
-  unset(GCF_SCRIPT)
+  set(GCF_GIT_DIR_SCRIPT ${GCF_GIT_HOOKS_DIR}/git-cmake-format.py)
+  if((EXISTS ${GCF_GIT_DIR_SCRIPT}) AND (EXISTS ${GCF_GIT_HOOKS_DIR}/pre-commit))
```

Comment:
if the worktree has newer GCF_GIT_DIR_SCRIPT, it will not be updated.
and the current ginkgo one dummy is wrong.

Reply:
I don't think you can compare these just by time, since the worktree version will always be newer (all files in it are newer than in the root repository). I could add a cmake option to force overwriting the files, but I'm not sure what the default value should be. Should we always overwrite by default?

Reply:
I thought to compare the git commit to check which commit is newer.
overwriting or ignoring should be the similar option for me

Reply:
I've added an option which defaults to `OFF`, but I intend to set it to `ON` from ginkgo. Comparing the commits sounds quite complicated.

Reply:
ON is good. when update the branch, it will also update the hook